File: service/src/main/java/com/github/goldy1992/mp3player/service/library/content/observers/MediaStoreObservers.java
package com.github.goldy1992.mp3player.service.library.content.observers;

import com.github.goldy1992.mp3player.service.MediaPlaybackService;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class MediaStoreObservers {

    private List<MediaStoreObserver> mediaStoreObserversList;

    @Inject
    public MediaStoreObservers(AudioObserver audioObserver) {
        this.mediaStoreObserversList = new ArrayList<>();
        this.mediaStoreObserversList.add(audioObserver);
    }

    public void init(MediaPlaybackService mediaPlaybackService) {
        for (MediaStoreObserver mediaStoreObserver : mediaStoreObserversList) {
            mediaStoreObserver.init(mediaPlaybackService);
        }
        registerAll();
    }

    public void registerAll() {
        for (MediaStoreObserver mediaStoreObserver : mediaStoreObserversList) {
            mediaStoreObserver.register();
        }
    }

    public void unregisterAll() {
        for (MediaStoreObserver mediaStoreObserver : mediaStoreObserversList) {
            mediaStoreObserver.unregister();
        }
    }
}

File: service/src/main/java/com/github/goldy1992/mp3player/service/library/content/observers/MediaStoreObservers.kt
package com.github.goldy1992.mp3player.service.library.content.observers

import androidx.media3.session.MediaLibraryService.MediaLibrarySession
import dagger.hilt.android.scopes.ServiceScoped
import javax.inject.Inject

@ServiceScoped
class MediaStoreObservers @Inject constructor(audioObserver: AudioObserver) {
    private val mediaStoreObserversList: MutableList<MediaStoreObserver>
    fun init(mediaLibrarySession: MediaLibrarySession) {
        for (mediaStoreObserver in mediaStoreObserversList) {
            mediaStoreObserver.init(mediaLibrarySession)
        }
        registerAll()
    }

    fun registerAll() {
        for (mediaStoreObserver in mediaStoreObserversList) {
            mediaStoreObserver.register()
        }
    }

    fun unregisterAll() {
        for (mediaStoreObserver in mediaStoreObserversList) {
            mediaStoreObserver.unregister()
        }
    }

    init {
        mediaStoreObserversList = ArrayList()
        mediaStoreObserversList.add(audioObserver)
    }
}